src/seedu/addressbook/data/AddressBook.java

```diff
@@ -22,13 +24,15 @@
 
     private final UniquePersonList allPersons;
     private final UniqueTagList allTags; // can contain tags not attached to any person
+    private ArrayList<Tagging> taggings; // keeps track of all the tags added and deleted for the current session
 
     /**
      * Creates an empty address book.
      */
     public AddressBook() {
         allPersons = new UniquePersonList();
         allTags = new UniqueTagList();
+        taggings = new ArrayList<>();
```

Comment:
There are two AdressBook constructors. Don't you need to initialize the tagging class in both place?